crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2209,8 +2209,25 @@ impl<'db> TypeInferenceBuilder<'db> {
         } = unary;
 
         match (op, self.infer_expression(operand)) {
+            (UnaryOp::UAdd, Type::IntLiteral(value)) => Type::IntLiteral(value),
             (UnaryOp::USub, Type::IntLiteral(value)) => Type::IntLiteral(-value),
+            (UnaryOp::Invert, Type::IntLiteral(value)) => Type::IntLiteral(-(value + 1)),
+
+            (UnaryOp::UAdd, Type::BooleanLiteral(bool)) => {
+                let value = i64::from(bool);
+                Type::IntLiteral(value)
+            }
+            (UnaryOp::USub, Type::BooleanLiteral(bool)) => {
+                let value = i64::from(bool);
+                Type::IntLiteral(-value)
+            }
+            (UnaryOp::Invert, Type::BooleanLiteral(bool)) => {
+                let value = i64::from(bool);
+                Type::IntLiteral(-(value + 1))
+            }
```

Comment:
I'm not sure this needs to be done now (and probably it actually _shouldn't_ be done now), but at some point we'll want to emit some kind of diagnostic here. It's currently deprecated at runtime. Whether it will remain deprecated is an [open question](https://discuss.python.org/t/bool-deprecation/62232), but it's nonetheless true that typeshed [decorates `bool.__invert__` with `@deprecated`](https://github.com/AlexWaygood/typeshed/blob/9f033bf439e064917e4b235b8a09b7530f182516/stdlib/builtins.pyi#L927-L928) in its `builtins` stub, which means that (once we support `@deprecated`) we'll emit a diagnostic when users attempt to invert a non-literal boolean value. For consistency, we'll want to make sure we emit the same diagnostic for literal `bool`s as we do for non-literal `bool`s, but I'm not sure how we want to do that. (Should we check the typeshed `bool` stub for each method when we're doing type inference on literal `bool`s, in case the `bool` method is decorated with `@deprecated` or similar? Or should we copy out the same deprecation method here, and hope that we can manually keep them in sync? Curious for @carljm's thoughts!)

TL;DR: let's add a TODO comment noting that eventually this should probably cause us to emit a diagnostic due to the deprecation!

Reply:
I think we'll probably want to get the deprecation message from calling the typeshed method, once we support `@deprecated`.